addons/binding/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/api/manager/EnergenieApiManagerImpl.java

```diff
@@ -0,0 +1,210 @@
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put(DeviceConstants.DEVICE_ID_KEY, id);
+        // With this parameter we exclude the historical usage data ("1" - to include, "0" - to exclude)
+        // The historical data is from no importance to the user,
+        // it consists of the number of events for each hour from the last 24 hours.
+        parameters.put(DeviceConstants.SUBDEVICE_INCLUDE_USAGE_DATA, 0);
+        return execute(CONTROLLER_SUBDEVICES, ACTION_SHOW, parameters);
+    }
+
+    @Override
+    public JsonObject updateSubdevice(int id, String label) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put(DeviceConstants.DEVICE_ID_KEY, id);
+        parameters.put(DeviceConstants.DEVICE_LABEL_KEY, label);
+        return execute(CONTROLLER_SUBDEVICES, ACTION_UPDATE, parameters);
+    }
+
+    @Override
+    public JsonObject unregisterSubdevice(int id) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put(DeviceConstants.DEVICE_ID_KEY, id);
+        return execute(CONTROLLER_SUBDEVICES, ACTION_DELETE, parameters);
+    }
+
+    private JsonObject execute(String controller, String action) {
+        return execute(controller, action, new HashMap<>());
+    }
+
+    private JsonObject execute(String controller, String action, Map<String, Object> content) {
+
+        Properties httpHeaders = new Properties();
+
+        // Set basic authentication header
+        String authData = configuration.getUserName() + ":" + configuration.getPassword();
+        String basicAuthentication = "Basic " + Base64.getEncoder().encodeToString(authData.getBytes());
+        basicAuthentication = basicAuthentication.trim();
+        httpHeaders.setProperty("Authorization", basicAuthentication);
+
+        // Convert request the content to JSON
+        Gson gson = new Gson();
+        String jsonContent = gson.toJson(content);
+        InputStream requestContent = IOUtils.toInputStream(jsonContent);
+        ContentResponse contentResponse = null;
+        String responseBody = null;
+
+        try {
+            // all requests are sent using POST method, as it is recommended from the API documentation
+            contentResponse = restClient.sendRequest(controller + "/" + action, RestClient.DEFAULT_HTTP_METHOD,
+                    httpHeaders, requestContent, RestClient.CONTENT_TYPE);
+
+            byte[] rawResponse = contentResponse.getContent();
+            String encoding = contentResponse.getEncoding() != null
+                    ? contentResponse.getEncoding().replaceAll("\"", "").trim() : "UTF-8";
+            responseBody = new String(rawResponse, encoding);
+        } catch (IOException e) {
+            logger.error("Request execution failed: ", e);
+            String failedUrl = restClient.getBaseURL() + controller + "/" + action;
+            failingRequestHandler.handleIOException(failedUrl, e);
```

Comment:
It feels inconsistent, it could be fine, but it seems now that some handler methods have the responsibility to log while others have not. 

It could be explained by the fact that there is some related information which is not available in the handler anymore and that is the reason that sometimes logging there would be too late.